### ISSUE TITLE
Using as es6 module in chrome extension

### ISSUE THREAD

**Issue by CubeOfCheese** [remusao/tldts#855]
Hello, I am trying to use this as an es6 module like so `import { parse } from "tldts";`
The browser extension is not a node.js project (but we are using npm just so we can use eslint) so I'm trying to understand how to import it. 

Currently receiving this error
```
Uncaught TypeError: Failed to resolve module specifier "tldts". Relative references must start with either "/", "./", or "../".
```
Which makes perfect sense except I don't know where in the file path the module specifier should be

**Comment by CubeOfCheese** [comment]
I supposed I could use browserify or something of the like but I'd rather not at the moment and since your readme said it supported es6 import I assumed it would work properly

**Comment by remusao** [comment]
Hi @CubeOfCheese,

I tried locally in a simple node.js project locally to check the es6 bundle and it seemed to work:
```sh
$ npm init
$ npm install --save tldts
```

Then I put the following in `index.mjs` (for es6 support):
```js
import { parse } from 'tldts';

console.log(parse('https://github.com'));
```

And running `node index.mjs` displayed:
```js
{
  domain: 'github.com',
  domainWithoutSuffix: 'github',
  hostname: 'github.com',
  isIcann: true,
  isIp: false,
  isPrivate: false,
  publicSuffix: 'com',
  subdomain: ''
}
```

Can you tell me more about your particular setup? How do you bundle your extension? Could it be that you need to resolve to the full path to bundle (maybe something like: `./node_modules/tldts/dist/index.esm.min.js`) and also bundle `node_modules` in your extension archive? An alternative could be to use a bundler such a rollup to produce a single file for your background and content scripts (I have a simple extension using this tool here: https://github.com/remusao/blockrz/blob/master/rollup.config.ts).

Best,
Rémi

**Comment by CubeOfCheese** [comment]
Hey Rémi thanks for getting back to me. We can't use `node index.mjs` as a way to run the extension.
In order to submit the extension we have to zip up the whole Extension directory.

Relevant parts of my file structure
```
.
├── Extension
│   ├── icons
│   │   ├── ethicli-128.png
│   │   ...
│   ├── images
│   │   ├── badge-dark.svg
│   │   ...
│   ├── stylesheets
│   │   ├── reset.css
│   │   ...
│   ├── background.html
│   ├── background.js
│   ├── content.js
│   ├── manifest.json
│   ├── popup-scripts
│   ├── popup.html
│   ├── popup.js
│   ├── popupNoRating.html
│   ├── popupNotShop.html
│   ├── popupOptin.html
│   ├── welcome-script.js
│   └── welcome.html
├── node_modules
│   ├── tldts
│   │   ├── CHANGELOG.md
│   │   ├── LICENSE
│   │   ├── README.md
│   │   ├── bin
│   │   │   └── cli.js
│   │   ├── dist
│   │   │   ├── cjs
│   │   │   │   ├── index.js
│   │   │   │   ├── index.js.map
│   │   │   │   ├── src
│   │   │   │   │   ├── data
│   │   │   │   │   │   ├── trie.js
│   │   │   │   │   │   └── trie.js.map
│   │   │   │   │   ├── suffix-trie.js
│   │   │   │   │   └── suffix-trie.js.map
│   │   │   │   └── tsconfig.tsbuildinfo
│   │   │   ├── es6
│   │   │   │   ├── index.js
│   │   │   │   ├── index.js.map
│   │   │   │   ├── src
│   │   │   │   │   ├── data
│   │   │   │   │   │   ├── trie.js
│   │   │   │   │   │   └── trie.js.map
│   │   │   │   │   ├── suffix-trie.js
│   │   │   │   │   └── suffix-trie.js.map
│   │   │   │   └── tsconfig.bundle.tsbuildinfo
│   │   │   ├── index.esm.min.js
│   │   │   ├── index.esm.min.js.map
│   │   │   ├── index.umd.min.js
│   │   │   ├── index.umd.min.js.map
│   │   │   └── types
│   │   │       ├── index.d.ts
│   │   │       └── src
│   │   │           ├── data
│   │   │           │   └── trie.d.ts
│   │   │           └── suffix-trie.d.ts
│   │   └── package.json
│   ├── tldts-core
│   │   ├── CHANGELOG.md
│   │   ├── LICENSE
│   │   ├── README.md
│   │   ├── dist
│   │   │   ├── cjs
│   │   │   │   ├── index.js
│   │   │   │   ├── index.js.map
│   │   │   │   ├── src
│   │   │   │   │   ├── domain-without-suffix.js
│   │   │   │   │   ├── domain-without-suffix.js.map
│   │   │   │   │   ├── domain.js
│   │   │   │   │   ├── domain.js.map
│   │   │   │   │   ├── extract-hostname.js
│   │   │   │   │   ├── extract-hostname.js.map
│   │   │   │   │   ├── factory.js
│   │   │   │   │   ├── factory.js.map
│   │   │   │   │   ├── is-ip.js
│   │   │   │   │   ├── is-ip.js.map
│   │   │   │   │   ├── is-valid.js
│   │   │   │   │   ├── is-valid.js.map
│   │   │   │   │   ├── lookup
│   │   │   │   │   │   ├── fast-path.js
│   │   │   │   │   │   ├── fast-path.js.map
│   │   │   │   │   │   ├── interface.js
│   │   │   │   │   │   └── interface.js.map
│   │   │   │   │   ├── options.js
│   │   │   │   │   ├── options.js.map
│   │   │   │   │   ├── subdomain.js
│   │   │   │   │   └── subdomain.js.map
│   │   │   │   └── tsconfig.tsbuildinfo
│   │   │   ├── es6
│   │   │   │   ├── index.js
│   │   │   │   ├── index.js.map
│   │   │   │   ├── src
│   │   │   │   │   ├── domain-without-suffix.js
│   │   │   │   │   ├── domain-without-suffix.js.map
│   │   │   │   │   ├── domain.js
│   │   │   │   │   ├── domain.js.map
│   │   │   │   │   ├── extract-hostname.js
│   │   │   │   │   ├── extract-hostname.js.map
│   │   │   │   │   ├── factory.js
│   │   │   │   │   ├── factory.js.map
│   │   │   │   │   ├── is-ip.js
│   │   │   │   │   ├── is-ip.js.map
│   │   │   │   │   ├── is-valid.js
│   │   │   │   │   ├── is-valid.js.map
│   │   │   │   │   ├── lookup
│   │   │   │   │   │   ├── fast-path.js
│   │   │   │   │   │   ├── fast-path.js.map
│   │   │   │   │   │   ├── interface.js
│   │   │   │   │   │   └── interface.js.map
│   │   │   │   │   ├── options.js
│   │   │   │   │   ├── options.js.map
│   │   │   │   │   ├── subdomain.js
│   │   │   │   │   └── subdomain.js.map
│   │   │   │   └── tsconfig.bundle.tsbuildinfo
│   │   │   └── types
│   │   │       ├── index.d.ts
│   │   │       └── src
│   │   │           ├── domain-without-suffix.d.ts
│   │   │           ├── domain.d.ts
│   │   │           ├── extract-hostname.d.ts
│   │   │           ├── factory.d.ts
│   │   │           ├── is-ip.d.ts
│   │   │           ├── is-valid.d.ts
│   │   │           ├── lookup
│   │   │           │   ├── fast-path.d.ts
│   │   │           │   └── interface.d.ts
│   │   │           ├── options.d.ts
│   │   │           └── subdomain.d.ts
│   │   └── package.json
```

I want to use this as a module in background.js. I tried using `./node_modules/tldts/dist/index.esm.min.js` like you said but I got a file not found error which indicates to me that tldts must be within the extension directory. However I don't know exactly what that should look like. I'm looking into rollup.js right now. Thank you for the recommendation. We currently aren't using any kind of bundler.

**Comment by CubeOfCheese** [comment]
Okay I was able to get it working with rollup! Thank you. It adds like 30MB to the extension's memory usage which seems like quite a lot but maybe a necessary evil.

**Comment by remusao** [comment]
Happy to hear that it works now! 30MB seems like a lot of memory. Did you measure it with the Chrome devtools memory snapshot?

If you want to reduce the memory usage I could recommend you give a shot to `tldts-experimental` (don't mind the name too much...) which exposes the same API then `tldts`, but which is using a much more compact data structure below the hood (it's also faster).

**Comment by CubeOfCheese** [comment]
I did the memory measurement using Chrome's task manager.
Thanks for recommending tldts-experimental. That only adds 6MB which is much more reasonable. I tried out normal tldts again and it looks like I might have made a mistake before. It looks like it's only adding about 11MB.
In case you're curious these are the measurements
Original extension: 19MB
With tldts: 25MB
With tldts-experimental: 30.5MB

Thank you so much for making this package and for all your help!
